nomad/blocked_evals.go

```diff
@@ -477,6 +497,28 @@ func (b *BlockedEvals) UnblockClassAndQuota(class, quota string, index uint64) {
 	}
 }
 
+// UnblockNode finds any blocked evalution that's node specific (system jobs) and enqueues
+// it on the eval broker
+func (b *BlockedEvals) UnblockNode(nodeID string, index uint64) {
+	b.l.Lock()
+	defer b.l.Unlock()
+
+	evals, ok := b.system.NodeEvals(nodeID)
+
+	// Do nothing if not enabled
+	if !b.enabled || !ok || len(evals) == 0 {
+		return
+	}
+
+	// QUESTION is it dangerous to delete this first?
```

Comment:
I believe this is fine - but @preetapan should know best.  If we want to be extra safe, we may need to have `EnqueueAll` return error on failure.

Reply:
This should be fine and follows a similar pattern as other places in eval broker where we need to unblock evals and re-enqueue them. 

Currently `EnqueueAll` can silently not enqueue anything if this node lost leadership between before this was called to the point where it needs to enqueue. But if that's an expected case handled in this method called during leadership transitions - https://github.com/hashicorp/nomad/blob/913427428a43be42b4f23f7d7f1f27ab3f406665/nomad/leader.go#L290